docs: adding note about middlewares and slices (#2378)
pmndrs/zustand: docs/guides/slices-pattern.md

)
```

> Please, keep in mind you should apply middlewares only in the combined store and avoid applying them inside individual slices to prevent issues.

Reviewer: ```suggestion
Please keep in mind you should only apply middlewares in the combined store. Applying them inside individual slices can lead to unexpected issues.
```
